src/iroh.rs

```diff
@@ -474,159 +422,281 @@ impl VeilidIrohBlobs {
+            }
+        };
+
+        println!("Successfully retrieved tag for collection: {}", collection_name);
+
+        // Fetch collection data by its hash
+        let collection_entry = self.store.get(&collection_hash).await?;
+        let mut collection: HashMap<String, Hash> = if let Some(entry) = collection_entry {
+            let mut reader = entry.data_reader();
+            let collection_data: Bytes = reader.read_to_end().await?;
+            match from_slice(&collection_data) {
+                Ok(deserialized_collection) => deserialized_collection,
+                Err(err) => {
+                    println!("Error deserializing collection: {:?}", err);
+                    return Err(anyhow!("Failed to deserialize collection: {:?}", err));
+                }
+            }
+        } else {
+            HashMap::new()
+        };
+
+        // Add or update the file in the collection (HashMap)
+        collection.insert(path.clone(), file_hash.clone());
+
+        // Serialize the updated HashMap to CBOR
         let cbor_data = to_vec(&collection)?;
-        
-            
-        // Write the CBOR data to a temporary file
-        let temp_path = self.base_dir.join(format!("{}_updated.cbor", collection_name));
-        std::fs::write(&temp_path, &cbor_data)?;
 
-        // Convert the path to an absolute path
+        // Upload the updated collection data to the store
+        let temp_path = self.base_dir.join("updated_collection.cbor");
+        std::fs::write(&temp_path, &cbor_data)?;
         let absolute_temp_path = std::fs::canonicalize(&temp_path)?;
-
-        // Upload the updated collection
         let new_collection_hash = self.upload_from_path(absolute_temp_path).await?;
+
+        // Log: Print the new collection hash
+        println!("New collection hash: {:?}", new_collection_hash);
+
+        // Store the new collection hash with the tag
+        self.store_tag(&collection_name, &new_collection_hash).await?;
+
         
-        // Update the collection hash in the HashMap
-        {
-            let mut map = self.collection_hashes.write().await;
-            map.insert(collection_name, new_collection_hash);
+        // Introduce a small delay to ensure the tag is properly stored and synced
+        tokio::time::sleep(Duration::from_millis(100)).await;
+
+
+        // Verify the tag immediately after storing
+        let verified_hash = self.get_tag(&collection_name).await?;
+        if verified_hash == new_collection_hash {
+            println!("Tag verified for collection: {}", collection_name);
+        } else {
+            return Err(anyhow!("Tag verification failed for collection: {}", collection_name));
         }
     
         Ok(new_collection_hash)
     }
+    pub async fn get_file(&self, collection_name: &String, path: &String) -> Result<Hash> {
+        // Retrieve the current collection hash from the tag
+        let collection_hash = self.get_tag(&collection_name).await?;
```

Comment:
We should have a method which gets the collection to deduplicate effort. like `let entry = self.get_collection(name)` which will get the hash from the tag and decode the cbor.